dex/models/trades/linea/platforms/lynex_fusion_linea_base_trades.sql

```diff
@@ -0,0 +1,56 @@
+{{
+    config(
+    schema = 'lynex_fusion_linea'
+    , alias = 'base_trades'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['tx_hash', 'evt_index']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+WITH dexs AS
+(
+    SELECT
+        t.evt_block_number AS block_number
+        , t.evt_block_time AS block_time
+        , t.recipient AS taker
+        , cast(null as varbinary) as maker
+        , CASE WHEN t.amount0 < INT256 '0' THEN abs(t.amount0) ELSE abs(t.amount1) END AS token_bought_amount_raw -- when amount0 is negative it means trader_a is buying token0 from the pool
+        , CASE WHEN t.amount0 < INT256 '0' THEN abs(t.amount1) ELSE abs(t.amount1) END AS token_sold_amount_raw
+        , CASE WHEN t.amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        , CASE WHEN t.amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
+        , t.contract_address as project_contract_address
+        , t.evt_tx_hash AS tx_hash
+        , t.evt_index
+    FROM
+        {{ source('lynex_linea', 'AlgebraPool_evt_Swap') }} t
+    INNER JOIN
+        {{ source('lynex_linea', 'AlgebraFactory_evt_Pool') }} f
```

Comment:
We also need to add these sources into *sources.yml.
By the way we may use `uniswap_compatible_v3_trades` macro in this spell.

Reply:
ok just let me know if you need me to provide anything

Reply:
You can read the [README](https://github.com/duneanalytics/spellbook/tree/main/dex/models/trades#readme) here.

I will continue to complete this PR when I have time.

Reply:
I have made use of `uniswap_compatible_v3_trades` in the spell